src/site/content/en/blog/file-system-access/index.md

```diff
@@ -319,6 +319,44 @@ await directoryHandle.removeEntry('Abandoned Masterplan.txt');
 await directoryHandle.removeEntry('Old Stuff', { recursive: true });
 ```
 
+### Drag and drop integration
+
+The
+[HTML Drag and Drop interfaces](https://developer.mozilla.org/en-US/docs/Web/API/HTML_Drag_and_Drop_API)
+enable web applications to accept
+[dragged and dropped files](https://developer.mozilla.org/en-US/docs/Web/API/HTML_Drag_and_Drop_API/File_drag_and_drop)
+on a web page.
+During a drag and drop operation, dragged file and directory items are associated
+with file entries and directory entries respectively.
+The `DataTransferItem.getAsFileSystemHandle()` method returns a promise with a `FileSystemFileHandle` object
```

Comment:
FYI FileSystemFileHandle and FileSystemDirectoryHandle are not exposed to web developers, which wasn't obvious when reading this sentence. I'm not sure how to make it clear.

Reply:
If they're not exposed, why are they being described?

Reply:
I was wrong. They are exposed!

Reply:
@tomayac What do you think of using them in the example?

```diff
     if (item.kind === 'file') {
       const entry = await item.getAsFileSystemHandle();
-      if (entry.kind === 'directory') {
+      if (entry instanceof FileSystemDirectoryHandle) {
         handleDirectoryEntry(entry);
       } else {
         handleFileEntry(entry);
       }
     }
```

Reply:
Not needed, developers can just test the `kind`. This is what it's for.

Reply:
FWIW, this suggestion was only based on the double use of `kind` (one for item, one for entry) which are not the same objects.

Reply:
Acknowledged. It is, however, also consistent. The spec makes it an [enum](https://wicg.github.io/file-system-access/#enumdef-filesystemhandlekind) on purpose. It is definitely an unfortunate clash of an older and a newer API.

Reply:
@tomayac can you add some 'watch-out' language. I wouldn't make it an aside or anything like that. It would be good to alert developers of a potentially confusing overload term.

Reply:
Please see https://github.com/GoogleChrome/web.dev/pull/4196/commits/26a234401f6e3ffa69e15128cff7da85f39ed424.